pkg/sypgp/sypgp.go

```diff
@@ -612,6 +619,140 @@ func SearchPubkey(search, keyserverURI, authToken string) error {
 	return nil
 }
 
+//function to obtain the algorithm name for key encryption
+func getEncryptionAlgorithmName(n string) (string, error) {
+	algorithmName := ""
+
+	code, err := strconv.ParseInt(n, 10, 64)
+	if err != nil {
+		return "", err
+	}
+	switch code {
+
+	case 1, 2, 3:
+		algorithmName = "RSA"
+	case 16:
+		algorithmName = "Elgamal"
+	case 17:
+		algorithmName = "DSA"
+	case 18:
+		algorithmName = "Elliptic Curve"
+	case 19:
+		algorithmName = "ECDSA"
+	case 20:
+		algorithmName = "Reserved"
+	case 21:
+		algorithmName = "Diffie-Hellman"
+	}
+	return algorithmName, nil
+}
+
+//function to obtain a date format from linux epoch time
+func date(s string) string {
+	ret := "NULL"
+	if s == "" {
+		return ret
+	}
+	if s == "none" {
+		return s + "		  	       "
+	}
+
+	c, _ := strconv.ParseInt(s, 10, 64)
+	ret = fmt.Sprintf("%v", time.Unix(c, 0))
+
+	return strings.TrimSpace(ret)
+}
+
+//function to reformat key search output in machine readable format
+func reformatMachineReadableOutput(keyText string) string {
```

Comment:
I'm not sure this shouldn't return `(string, error)`. rather than just `string`. If `keyText` is malformed, the caller has no way to detect that a problem occurred.